### PR TITLE
[FIX] Fixes gnk ctd problems.

### LINKED ISSUE #2026: CTD support broken by new function _typeToString
GenericKnimeNodes expects `int` instead of `integer`as a type. When we replaced the hardcoded types [here](https://github.com/seqan/seqan/blob/966da62f2ba61867359cad18f4db2dbaf431243e/include/seqan/arg_parse/arg_parse_ctd_support.h#L394) with _typeToString() the `int` was replaced by `integer` ([here](https://github.com/seqan/seqan/blob/966da62f2ba61867359cad18f4db2dbaf431243e/include/seqan/arg_parse/arg_parse_argument.h#L234) )

This causes a failure in generating KNIME nodes. 

And one more thing 
`version-check` parameter has a `bool` type but the restrictions are `1,ON,TRUE,T,YES,0,OFF,FALSE,F,NO`.

This is also causing a problem. We either have to make it a `string` type or restrict the "restrictions" to TRUE and FALSE. (See [`getRestrictions`](https://github.com/seqan/seqan/blob/966da62f2ba61867359cad18f4db2dbaf431243e/include/seqan/arg_parse/arg_parse_ctd_support.h#L164) )

### PR DESCRIPTION
**Pull request by temehi**
This fixes #2026

### CODE REVIEW

File: include/seqan/arg_parse/arg_parse_argument.h [hunk 1]
```diff
@@ -244,27 +244,27 @@ inline std::string _typeToString(ArgParseArgument const & me)
...
 
     case ArgParseArgument::INPUT_DIRECTORY:
-        typeName = "input_directory";
+        typeName = "input-directory";
         break;
 
     case ArgParseArgument::OUTPUT_DIRECTORY:
-        typeName = "output_directory";
+        typeName = "output-directory";
         break;
```
Comment (h-2) [inline include/seqan/arg_parse/arg_parse_argument.h]: don't these changes affect something other than the CTD stuff?
@smehringer  what is your opinion on this?

Comment (h-2) [inline include/seqan/arg_parse/arg_parse_argument.h]: also, can ctd handle the directories?

Comment (temehi) [inline include/seqan/arg_parse/arg_parse_argument.h]: @h-2 take a look at this commit https://github.com/seqan/seqan/commit/9ca8c89eac64fe7afd058471fd9bcdc174076841

It was this way before.

Comment (h-2) [inline include/seqan/arg_parse/arg_parse_argument.h]: ok, but what about the directories, thats new, do you know if it works?

**Comment by h-2** [comment]
@smehringer do you approve of these changes? Then I will merge!

File: include/seqan/arg_parse/arg_parse_ctd_support.h [hunk 2]
```diff
@@ -289,15 +287,69 @@ inline std::string _getManual(ArgumentParser const & me)
...
+inline std::string _toValidGKNTypeSpecifier(ArgParseArgument const & me)
 {
-    if (type == "integer")
-        return "int";
-    return type;
+    std::string typeName = "";
+
+    switch (me._argumentType)
+    {
+    case ArgParseArgument::BOOL:
```
Comment (h-2) [inline include/seqan/arg_parse/arg_parse_ctd_support.h]: please indent the case labels properly!

File: include/seqan/arg_parse/arg_parse_ctd_support.h [hunk 3]
```diff
@@ -289,15 +287,69 @@ inline std::string _getManual(ArgumentParser const & me)
...
+        break;
+
+    case ArgParseArgument::OUTPUT_DIRECTORY:
+        typeName = "output-prefix";
+        break;
+
+    default:
+        typeName = "unknown";
+        break;
+    }
```
Comment (h-2) [inline include/seqan/arg_parse/arg_parse_ctd_support.h]: please specify only those cases that differ and make `default` call the other function to handle unchanged values.

**Comment by h-2** [comment]
You did not address my second comment.

**Comment by h-2** [comment]
Ah, sorry. I looked in the wrong place. I see that you did address it! However, don't change the other function, if you are only changing whitespaces. Never mix whitespace changes with functional changes!

Also you can greatly simplifiy your function by just return the string literal instead of assign to the local variable first...

**Comment by temehi** [comment]
@h-2. Sorry for my late response. 
Wouldn't that apply to the original function as well?

Should I just change the new one as you said and remove my remove the white space changes from the original function? Or should I also improve it?

**Comment by h-2** [comment]
yes, it would apply to the original function, but please just change it in the new one, don't change the old function at all, if you are not changing the behaviour.
If you want to change the old function, please make sure that you have a clean commit history, i.e. a single commit for your functional changes, and single commit for formatting/whitespace changes.

**Pull request closed without merging**